src/dune_engine/build_system.ml

```diff
@@ -2338,34 +2338,30 @@ let run f =
   in
   let f () =
     let* () = Handler.report_build_event t.handler Start in
-    let build_end_status_reported = ref false in
+    let current_build_end_status = ref Handler.Finish in
```

Comment:
Do we even need this reference? Feels like we could do a `List.fold_left` on the `exns` in the `| Error exns ->` branch.

Reply:
For some reason I had assumed that the `exns` reported at the end was not exhaustive. Fixed.